build/webpack/_development.js

```diff
@@ -41,12 +42,33 @@ export default (webpackConfig) => {
           imports   : ['react'],
           locals    : ['module']
         }
-        loader.query.env.development.plugins[0][1].transforms
-          .push(reactTransformHmr)
+
+        loader.query.env.development.plugins =
+          loader.query.env.development.plugins.map(plugin => {
```

Comment:
i'm sorry i wrote that really quickly didn't mean for it to come off like that!

yea i just meant it's a verbose solution for something that was one line. i hate that the mutation is hard-coded the way that it is, but it's so easy to understand the former way, and so complicated the latter. 

i totally understand on flexibility, but i think simplicity trumps in this instance. i'll leave it up to @davezuko  

if we leave it hard-coded, i agree we should make it a convention that react-transform must the first plugin and add a comment in _base webpack config